### PR TITLE
fix(tags): onChange prop type definition

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Bug

#### The following changes where made:

Update the `onChange` prop type definition so each tag is only a `string`, not a `string | null | undefined`

### CODE REVIEW

File: src/components/Tags/index.tsx [hunk 1]
```diff
@@ -145,9 +145,14 @@ const Tags: VoidFunctionComponent<TagsProps> = ({
   const inputRef = useRef<HTMLInputElement>(null)
 
   const dispatchOnChange = (newState: TagsProp) => {
-    const changes = newState.map(tag =>
-      typeof tag === 'object' ? tag?.label : tag,
-    )
+    function isTagDefined(tag: string | null | undefined): tag is string {
```
Comment (chambo-e) [inline src/components/Tags/index.tsx:148]: I don't understand why this is necessary ? 🤔

Comment (QuiiBz) [reply src/components/Tags/index.tsx:148]: Typescript won't infer the type with a normal filter without specifying the `tag is string` part:

Screenshot 2021-12-23 at 10 14 51

Comment (chambo-e) [reply src/components/Tags/index.tsx:148]: What if you move away from VoidFunctionComponent ?

Comment (QuiiBz) [reply src/components/Tags/index.tsx:148]: Same issue, if I try to manually type the `filter`, typescript effectively tells me to use a type predicate, so I don't think there's another way to solve this:

Screenshot 2021-12-23 at 10 23 01

Comment (chambo-e) [reply src/components/Tags/index.tsx:148]: You also need to set 

```ts
type TagsProp = (
  | string
  | { label: string; index: string }
)[]

```

Comment (QuiiBz) [reply src/components/Tags/index.tsx]: Awesome!

**Review by chambo-e — APPROVED** [review]

**Review by matthprost — APPROVED** [review]

**Pull request merged**